livingatlas/pipelines/src/main/java/au/org/ala/pipelines/beam/SamplingPipeline.java

```diff
@@ -0,0 +1,232 @@
+import au.org.ala.pipelines.options.SamplingPipelineOptions;
+import au.org.ala.pipelines.util.SamplingUtils;
+import au.org.ala.pipelines.util.VersionInfo;
+import au.org.ala.sampling.Layer;
+import au.org.ala.sampling.SamplingService;
+import au.org.ala.utils.ALAFsUtils;
+import au.org.ala.utils.CombinedYamlConfiguration;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.beam.sdk.Pipeline;
+import org.apache.beam.sdk.PipelineResult;
+import org.apache.beam.sdk.coders.AvroCoder;
+import org.apache.beam.sdk.io.AvroIO;
+import org.apache.beam.sdk.io.TextIO;
+import org.apache.beam.sdk.transforms.*;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.hadoop.fs.FileSystem;
+import org.gbif.pipelines.common.beam.metrics.MetricsHandler;
+import org.gbif.pipelines.common.beam.options.PipelinesOptionsFactory;
+import org.gbif.pipelines.common.beam.utils.PathBuilder;
+import org.gbif.pipelines.core.factory.FileSystemFactory;
+import org.gbif.pipelines.core.utils.FsUtils;
+import org.gbif.pipelines.io.avro.IndexRecord;
+import org.gbif.pipelines.io.avro.SampleRecord;
+import org.slf4j.MDC;
+
+/**
+ * A pipeline that exports a unique set of coordinates for a dataset into CSV. This pipeline can
+ * only be ran after the {@link ALAVerbatimToInterpretedPipeline} has been ran as it relies on the
+ * output of the LocationTransform.
+ *
+ * <p>In addition, this pipeline will check configured sampling service for newly available layers.
+ */
+@Slf4j
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class SamplingPipeline {
+
+  public static void main(String[] args) throws Exception {
+    VersionInfo.print();
+    CombinedYamlConfiguration conf = new CombinedYamlConfiguration(args);
+    String[] combinedArgs = conf.toArgs("general", "sampling");
+    SamplingPipelineOptions options =
+        PipelinesOptionsFactory.create(SamplingPipelineOptions.class, combinedArgs);
+    MDC.put("datasetId", options.getDatasetId());
+    MDC.put("attempt", options.getAttempt().toString());
+    MDC.put("step", "SAMPLING");
+    PipelinesOptionsFactory.registerHdfs(options);
+    run(options);
+    // FIXME: Issue logged here: https://github.com/AtlasOfLivingAustralia/la-pipelines/issues/105
+    System.exit(0);
+  }
+
+  public static void run(SamplingPipelineOptions options) throws Exception {
+
+    FileSystem fs =
+        FileSystemFactory.getInstance(options.getHdfsSiteConfig(), options.getCoreSiteConfig())
+            .getFs(options.getTargetPath());
+
+    log.info("Checking for new layers in the system");
+    SamplingService samplingService = SamplingUtils.initSamplingService(options.getBaseUrl());
+
+    boolean newLayersAvailable = newLayersAddedSinceLastSample(samplingService, options, fs);
```

Comment:
Should we check for dropped layers too? If a layer been deleted, do we need to remove that from the sampling?

Reply:
Thanks @sadeghim. I agree it would be nice to handle this. Ive raised this issue https://github.com/AtlasOfLivingAustralia/la-pipelines/issues/290 so we can handle it in future PR.